staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5073,6 +5073,10 @@ type NodeStatus struct {
 	// Note: This field is declared as mergeable, but the merge key is not sufficiently
 	// unique, which can cause data corruption when it is merged. Callers should instead
 	// use a full-replacement patch. See https://pr.k8s.io/79391 for an example.
+	// Consumers should assume that addresses can change during the
+	// lifetime of a Node. However, there are some exceptions where this may not
```

Comment:
may not possible because we don't allow it (e.g., validation?). Or may not possible to drive changes to host pods?.

Reply:
@kal I think that's covered? eg Node addresses may change over the lifetime of the node, but HostNet containers won't necessarily change their IP if the Node address changes after they have started. Also downward API to pods may not be updated inside the container after the pod has started.

Reply:
The wording is not clear if you ask me. What we want to say that node address changes don't trickle to host pods. The way it is worded right now "may possible" refers to node address not pods.

I am ok with current wording though i think we should improve it.

Reply:
It's ugly that we don't update pod IP in these cases, but it's SUCH a corner case...